sdk/iot/hub/src/az_iot_hub_client_c2d.c

```diff
@@ -12,28 +12,38 @@
 
 static const az_span c2d_topic_prefix = AZ_SPAN_LITERAL_FROM_STR("devices/");
 static const az_span c2d_topic_suffix = AZ_SPAN_LITERAL_FROM_STR("/messages/devicebound/");
+static const uint8_t null_terminator = '\0';
```

Comment:
Should be marked for post-public-preview that we should consolidate all these constants in one file.